Dockerfile

```diff
@@ -1,15 +1,23 @@
+ARG ALPINE_VERSION=3.12
+ARG NODE_VERSION=15
+ARG NGINX_VERSION=1.19.6
+
 # multistage - builder image
-FROM node:alpine AS builder
+FROM node:${NODE_VERSION}-alpine${ALPINE_VERSION} AS builder
 WORKDIR /app/Lightspeed-react
-COPY . .
+# Layer caching for node modules
+COPY package.json package-lock.json ./
 RUN npm install
-
-# configure ip, hardcoded to webrtc container address (8080) for now
-RUN sed -i "s|stream.gud.software|localhost|g" public/config.json
+COPY . .
 
 # build it
 RUN npm run build
 
 # runtime image
-FROM nginx:stable
-COPY --from=builder /app/Lightspeed-react/build /usr/share/nginx/html
+FROM nginx:${NGINX_VERSION}-alpine
+ENV WEBSOCKET_HOST=localhost
+ENV WEBSOCKET_PORT=8080
+
+COPY --chown=1000 docker/entrypoint.sh /docker-entrypoint.d/entrypoint.sh
+COPY --chown=1000 docker/config.json.template /config.json.template
+COPY --from=builder --chown=1000 /app/Lightspeed-react/build /usr/share/nginx/html
```

Comment:
You should add a `EXPOSE` at the end for documentation sake.

Reply:
I've added ports 8000 (websocket) and 80, assuming that's the one we use for Nginx.

Reply:
I think `lightspeed-react` container's job is only the react/nginx website on port on (container) port 80. Isn't `lightspeed-ingest` or `lightspeed-webrtc` for websocket?

Reply:
Yes you're totally right, I'll remove that websocket port 👍